### PR TITLE
MCM - 11

### PR DESCRIPTION
**Pull request by taniarodriguez**
Create entity medical center, create CRUD and create test to all entities

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -4886,6 +4886,11 @@ neo-async@^2.5.0, neo-async@^2.6.1:
   resolved "https://registry.yarnpkg.com/neo-async/-/neo-async-2.6.1.tgz#ac27ada66167fa8849a6addd837f6b189ad2081c"
   integrity sha512-iyam8fBuCUpWeKPGpaNMetEocMt364qkCsfL9JuhjXX6dRnguRVOfk2GZaDpPjcOKiiXCPINZC1GczQ7iTq3Zw==
 
+nest@^0.1.6:
```
Comment (EngAndres) [inline yarn.lock]: este archivo debería estar en tu .gitignore, y no ser actualizado mediante commits.

File: src/entities/medicalCenter.entity.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Column({type: 'varchar', length: 30, unique: true, nullable:false})
+    @IsDefined()
+    @Length(2,30)
+    name: string;
+    
+    @Column({type: 'varchar', length: 30, unique:true, nullable:false})
+    @IsDefined()
+    @Length(2,30)
+    document: string;
```
Comment (EngAndres) [inline src/entities/medicalCenter.entity.ts]: document in medical center? are you sure this one is the right name?

File: tests/unit/diagnoses/diagnoses.entity.test.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+it('diagnose all columns', async() => {
+    expect(await validate(diagnose)).toEqual([]);
+});
+
+it('diagnose without description', async() => {
+    expect(await validate(diagnose2)).toEqual([]);
+});
+
+it('Field ID is undefined', async() =>{
```
Comment (EngAndres) [inline tests/unit/diagnoses/diagnoses.entity.test.ts]: Test with: id: null, id: undefined.
Just for a couple of fields.

File: tests/unit/medicalCenter/medicalCenter.entity.test.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+it('Field Phone is undefined', async() => {
+    medical.phone = '';
+    expect(Error[0]).toBe(undefined);
+});
+
+it('Field Document is undefined', async() => {
+    medical.document = '';
+    expect(Error[0]).toBe(undefined);
+});
+
```
Comment (EngAndres) [inline tests/unit/medicalCenter/medicalCenter.entity.test.ts:93]: remove this empty lines

File: src/medical-center/medical-center.controller.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  
+
+*/
+import { Controller, Body, Param, Get, Delete,Post, Put, HttpException } from '@nestjs/common';
+import { Utilities} from '../Utilities';
+import { MedicalCenterService } from './medical-center.service';
+import { MedicalCenter } from '../entities/medicalCenter.entity';
+
+@Controller('medical-center')
+export class MedicalCenterController extends Utilities {
```
Comment (EngAndres) [inline src/medical-center/medical-center.controller.ts]: este "extends" no debería ser necesario, con el solo import debería bastar

File: src/medical-center/medical-center.controller.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    @Get(':idCenter')
+    async getOne(@Param('idCenter') idCenter:string):Promise<MedicalCenter>{
+        return this.medicalServices.findOne(idCenter);
+    }
+
+    @Post()
+    async create (@Body() medicalData: MedicalCenter): Promise<MedicalCenter>{
+        Utilities.checkParameterExistence(medicalData.address);
+        Utilities.checkParameterExistence(medicalData.document);
+        Utilities.checkParameterExistence(medicalData.isActive);
```
Comment (EngAndres) [inline src/medical-center/medical-center.controller.ts]: this field must not be mandatory...when you create it, this field could have a default value as "true"

File: src/medical-center/medical-center.controller.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        Utilities.checkParameterExistence(medicalData.name);
+        Utilities.checkParameterExistence(medicalData.phone);
+
+        return this.medicalServices.save(medicalData);
+    }
+
+    @Put(':id')
+    async update(@Param('id') id:string, @Body() medicalData: MedicalCenter): Promise<MedicalCenter>{
+        const medical: MedicalCenter = await this.medicalServices.findOne(id);
+        medicalData.id = id;
```
Comment (EngAndres) [inline src/medical-center/medical-center.controller.ts]: this is a mistake, you cannot update the id :O :O

File: src/medical-center/medical-center.controller.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        return this.medicalServices.save(medicalData);
+    }
+
+    @Delete(':id')
+    async delete(@Param('id') id:string): Promise<MedicalCenter>{
+        const medical: MedicalCenter = await this.medicalServices.findOne(id);
+        medical.isActive = false;
+        return this.medicalServices.save(medical);
+    }
+
```
Comment (EngAndres) [inline src/medical-center/medical-center.controller.ts]: remove these empty fields

File: src/medical-center/medical-center.controller.ts (hunk 7 above)
Comment (EngAndres) [reply src/medical-center/medical-center.controller.ts]: @Put(':id')
    async update(@Param('id') id:string, @Body() medicalData: MedicalCenter): Promise<MedicalCenter>{
        const medical: MedicalCenter = await this.medicalServices.findOne(id);
        //not all fields fields can be updated, please, be more careful with that
        Utilities.checkParameterExistence(medicalData.address);
        Utilities.checkParameterExistence(medicalData.document);
        Utilities.checkParameterExistence(medicalData.name);
        Utilities.checkParameterExistence(medicalData.phone);
       //you better update the same object you get from DB
       medical.phone = medicalData.phone;
        return this.medicalServices.save(medicalData);
    }

File: tests/unit/event/event.controller.test.ts [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+        return expect(controller.getOne("123")).resolves.toBe(event);
+    });
+
+
+    it.skip('GetOneElementNotExist', async() =>{
+        jest.spyOn(repository,'findOne').mockResolvedValueOnce(undefined);
+        return expect(controller.getOne("1234")).rejects.toThrowError(NotFoundException);
+    });
+
+});
```
Comment (EngAndres) [inline tests/unit/event/event.controller.test.ts:103]: you need to add test for update and delete

**Review by EngAndres — CHANGES_REQUESTED** [review]

File: yarn.lock (hunk 1 above)
Comment (taniarodriguez) [reply yarn.lock]: Done!

File: src/entities/medicalCenter.entity.ts (hunk 2 above)
Comment (taniarodriguez) [reply src/entities/medicalCenter.entity.ts]: Changed name to "numberDocument" and add field "typeDocument"

File: tests/unit/medicalCenter/medicalCenter.entity.test.ts (hunk 4 above)
Comment (taniarodriguez) [reply tests/unit/medicalCenter/medicalCenter.entity.test.ts:93]: Done

File: src/medical-center/medical-center.controller.ts (hunk 5 above)
Comment (taniarodriguez) [reply src/medical-center/medical-center.controller.ts]: Done

File: src/medical-center/medical-center.controller.ts (hunk 6 above)
Comment (taniarodriguez) [reply src/medical-center/medical-center.controller.ts]: Ok, changed it

File: src/medical-center/medical-center.controller.ts (hunk 8 above)
Comment (taniarodriguez) [reply src/medical-center/medical-center.controller.ts]: Ok

File: src/medical-center/medical-center.controller.ts (hunk 7 above)
Comment (taniarodriguez) [reply src/medical-center/medical-center.controller.ts]: Changed it!

File: tests/unit/diagnoses/diagnoses.entity.test.ts (hunk 3 above)
Comment (taniarodriguez) [reply tests/unit/diagnoses/diagnoses.entity.test.ts]: Added tests for all required fields (empty, undefined, null)

File: tests/unit/event/event.controller.test.ts (hunk 9 above)
Comment (taniarodriguez) [reply tests/unit/event/event.controller.test.ts:103]: Add test for necessary fields

**Review by EngAndres — APPROVED** [review]
So good!

**Pull request merged**